### PR TITLE
[Bug Fix] Returning timeseries_limit_metric in table viz get_data

### PR DESCRIPTION
**Pull request by michellethomas**
### CATEGORY

Choose one

- [x] Bug Fix
- [ ] Enhancement (new features, refinement)
- [ ] Refactor
- [ ] Add tests
- [ ] Build / Development Environment
- [ ] Documentation

### SUMMARY
[This](https://github.com/apache/incubator-superset/pull/9122) change update the table viz's get_data, but was missing returning the timeseries_limit_metric (sort_by). If a metric has a custom label and a sort by field, not returning the sort_by field doesn't show the metric.

### TEST PLAN
Tested a metric with a label and a sort by field

### ADDITIONAL INFORMATION

- [ ] Has associated issue:
- [ ] Changes UI
- [ ] Requires DB Migration.
- [ ] Confirm DB Migration upgrade and downgrade tested.
- [ ] Introduces new feature or API
- [ ] Removes existing feature or API

### REVIEWERS
@john-bodley

### CODE REVIEW

**Review by john-bodley — COMMENTED** [review]
@michellethomas apologies for the regression. Do you think it would be possible to augment the test case to add the timeseries limit metric?

**Comment by michellethomas** [comment]
@john-bodley also needed to fix a similar issue with the include_time field too. I added a test for the timeseries_limit_metric. I'll add a test for include_time in a follow up PR.

File: superset/viz.py [hunk 1]
```diff
@@ -575,15 +575,29 @@ def get_data(self, df: pd.DataFrame) -> VizData:
...
-        if not self.should_be_timeseries() and DTTM_ALIAS in df:
-            del df[DTTM_ALIAS]
-
+        non_percent_metric_columns = []
         # Transform the data frame to adhere to the UI ordering of the columns and
         # metrics whilst simultaneously computing the percentages (via normalization)
         # for the percent metrics.
-        non_percent_metric_columns = (
+
+        if not self.should_be_timeseries() and DTTM_ALIAS in df:
```
Comment (john-bodley) [inline superset/viz.py]: Maybe

```python
if DTTM_ALIAS in df:
    if self.should_be_timeseries():
        non_percent_metric_columns.append(DTTM_ALIAS)
    else:
        del df[DTTM_ALIAS]
```

File: superset/viz.py [hunk 2]
```diff
@@ -575,15 +575,29 @@ def get_data(self, df: pd.DataFrame) -> VizData:
...
+            non_percent_metric_columns.append(DTTM_ALIAS)
+
+        non_percent_metric_columns.extend(
             self.form_data.get("all_columns") or self.form_data.get("groupby") or []
-        ) + utils.get_metric_names(self.form_data.get("metrics") or [])
+        )
+
+        non_percent_metric_columns.extend(
+            utils.get_metric_names(self.form_data.get("metrics") or [])
+        )
```
Comment (john-bodley) [inline superset/viz.py]: How are lines 588 - 594 different to what was previously on lines 584 - 586?

Comment (michellethomas) [reply superset/viz.py]: I'm just extending them separately because it feels a bit more readable, rather than extending list + list.

**Review by john-bodley — APPROVED** [review]

**Pull request merged**